core/src/main/java/org/apache/camel/kafkaconnector/CamelSinkTask.java

```diff
@@ -122,7 +122,14 @@ public void put(Collection<SinkRecord> sinkRecords) {
             }
             exchange.getMessage().setHeaders(headers);
             exchange.getMessage().setBody(record.value());
-            LOG.debug("Sending {} to {}", exchange, LOCAL_URL);
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Sending exchange {} to {}", exchange.getExchangeId(), LOCAL_URL);
+            }
+            if (config.getBoolean(CamelSinkConnectorConfig.CAMEL_SINK_LOG_PAYLOAD_CONF)) {
+                LOG.info(TaskHelper.getExchangePayload(exchange));
```

Comment:
As @davsclaus pointed out in a comment, sensitive data can be in other places than the body and is not clear from the option name that, if enabled, you log at info level (I would have left it to debug and harmonized somehow with the previuse logging statement in order not to duplicate logging entries).

Reply:
Now log is on ConnectRecord, rather than Camel Exchange.